Enterprise-CMCS/eAPD
# updated assurance and compliance for mmis

Resolves #4473 

### Description
Updated the Assurance and Compliance components to handle MMIS data. Removed some old code that creates citations on the front end if they don't already exist in the APD. This isn't needed any more because all of the APDs in staging and prod have assurance and compliance. They get added when you create and new APD. And it would be extra complicated to update it to handle the new types. Also created storybook stories.


### Automated test cases written

| Given | When | Then | Type (jest, tap, cypress) |
| ----- | ---- | ---- | ------------------------- |
| a MMIS APD | the Assurances and Compliance page is loaded | it renders correctly | jest |
| a MMIS APD | the Assurances and Compliance Read Only page is loaded | it renders correctly | jest |
| a user on the Assurances and Compliance page | the user sets an IV&V compliance | an action is dispatched to update the APD | jest |
| a user on the Assurances and Compliance page | the user sets an IV&V justification | an action is dispatched to update the APD | jest | 

### Steps to manually verify this change

1. open a MMIS APD
2. navigate to the Assurances and Compliance page
3. verify that the citation match the ones from the ticket
4. verify that all of the citations have a url
5. verify that the urls are valid
6. select Yes
7. select No and a text box should open
8. enter text in the text box
9. select Yes and the text box should disappear
10. select No on a citation and don't enter any text in the text box
11. leave a citation empty
12. turn on Admin Check
13. verify that you see an error message under the empty text box
14. verify that you see an error message under the empty citation
15. verify that you see the errors in the Admin Check panel
16. verify that entering text and making a selection make the error message go away
17. verify that the error messages went away in the Admin Check panel
18. open a HITECH APD
19. navigate to the Assurances and Compliance page
20. verify that the citations are different than they were for the MMIS APD

### This pull request is ready to code review when

      tests, refactor

### This pull request is ready to test when


### This pull request is ready to review when the QA has


### This pull request can be merged when
